compose/cli/main.py

```diff
@@ -1312,3 +1321,51 @@ def build_exec_command(options, container_id, command):
     args += [container_id]
     args += command
     return args
+
+
+def has_container_with_state(containers, state):
+    for container in containers:
+        states = {
```

Comment:
I feel we can improve this by taking the instantiation outside of the loop and using lambdas, i.e.
```python
states = {
  'running': lambda c: c.is_running,
  'stopped': lambda c: not c.is_running,
 ...
}
if state not in states:
  raise UserError(...)
for container in containers:
  if states[state](container):
    return True
return False
```

Reply:
Also, it should be trivial to support `restarting` as a state filter as well.

Reply:
Done for both!